### ISSUE TITLE
Deployment doesn't fail when the environment is unavailable

### ISSUE THREAD

**Issue by toniedzwiedz** [adobe/aem-site-template-standard#33]
### Expected Behaviour

The Github workflow responsible for deploying an updated theme to AaCS fails when the instance is unavailable. It only succeeds when the artifact has been successfully deployed.

### Actual Behaviour

If deployment is attempted while the AEM instance is unavailable, the workflow is marked as successfully completed, which could be confusing under certain circumstances.

Under the hood, `npx aem-site-theme-builder deploy` makes an HTTP request to transfer the zip to AEM. If it receives a non-successful response code, the failure is silently ignored.

#### Steps to Reproduce

1. Create a site based on the AEM Site Template
2. Put it under version control on Github and configure Github Actions to automate deployments
3. Hibernate the Environment via the Developer Console
4. Trigger the _Build and deploy to Github artifacts_ workflow
5. Wait for the _Update theme config on AEM with artifact id_ workflow to kick off and inspect the log

#### Platform and Version

Site generated using `"@adobe/aem-site-theme-builder": "4.0.1"`

#### Logs taken while reproducing problem

From the output of the `update_theme_config` workflow kicked off while the relevant environment was hibernated. AEM responds with HTTP 503 but the workflow succeeds.

```
Run npx aem-site-theme-builder deploy
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  1269  100  1141  100   128   2616    293 --:--:-- --:--:-- --:--:--  2903
<!DOCTYPE html>
              <html lang="en">
                <head>
                  <meta charset="utf-8">
                  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
                  <meta name="description" content="AEM Cloud Service">
                  <title>AEM Cloud Service</title>
                  <link rel="stylesheet" href="https://errors.adobeaemcloud.com/src.status.css">
                  <link rel="icon" href="https://errors.adobeaemcloud.com/favicon.ico">
                </head>
                <body>
                  <noscript>AEM Cloud Service.</noscript>
                  <div id="root" status="503"
                        backend="devconsole"
                        healthy="hibernated"
                        rid="0"
                        statusApi="https://dev-console-ns-team-aem-cm-prd-n6712.ethos13-prod-nld2.dev.adobeaemcloud.com/api/releases/ns-team-aem-cm-prd-n6712/status/cm-p8802-e42804"
                  > </div>
                  <script src="https://errors.adobeaemcloud.com/src.status.js"> </script>
                </body>
              </html>
```

The information in the log is quite clear but there's no visual hint at the failure at first. It is only once we've tried to access the environment that we realised it had been hibernated automatically.

![successful runs that didn't actually deploy anything](https://user-images.githubusercontent.com/8336085/109177109-bec7fd80-7787-11eb-87dd-8eaf35af64dc.png). 

This is a trivial case in case of hibernation but it could behave in similar ways in case of unexpected problems with the environment and, depending on the development flow, some time-consuming and/or expensive tests could be started in a situation where the failure could be immediately obvious earlier. I think a fail-fast strategy would be a nice improvement here.

**Comment by vladbailescu** [comment]
The beta flow is no superseded by changes in https://github.com/adobe/aem-site-template-standard/pull/48

When the Quick Site Creation flow will be generally available, the build and deploy pipeline will be through Cloud Manager.
